### DISCUSSION TITLE
Benchmarks:matrixmultiplication

### DISCUSSION THREAD

**Discussion by yyssys** [sarchlab/mgpusim#55] [Q&A] [answered]
Hello author,  In the benchmark matrix multiplication, the parameter X is defined to represent the height of the first matrix. My understanding is that it represents the number of rows in the first matrix; Defined parameter Y: represents the width of the first matrix and the height of the second matrix. My understanding is that it represents the number of columns in the first matrix and the number of rows in the second matrix; The parameter Z is defined to represent the width of the second matrix, and my understanding is that it represents the number of columns in the second matrix. But in initializing the memory function, the parameter passed in when generating the second matrix is (Z, X), shouldn't I think it should be (Y, Z)? And in the function of constructing the matrix, the first parameter passed in is the width. This does not match the description of X, Y, and Z. I kindly request the author's guidance.
![Screenshot_2024-04-07-00-39-49-70_a252b927494330cdc2c8ba3b3f952e5e](https://github.com/sarchlab/mgpusim/assets/161021701/ff264d58-4290-48e9-95c0-8fdcbf232c58)
![IMG_20240407_002941](https://github.com/sarchlab/mgpusim/assets/161021701/83a2e035-05e7-44b6-82ad-764895d53b02)
![IMG_20240407_003126](https://github.com/sarchlab/mgpusim/assets/161021701/351a0c79-105f-449a-8694-cd14cf432030)
![IMG_20240407_003215](https://github.com/sarchlab/mgpusim/assets/161021701/92563d8c-f1c2-4b19-bf25-94223602689a)

**Comment by syifan** [comment, marked as answer]
It is actually better to use M, N, K as the parameter names, which are more widely accepted. 

<img width="589" alt="gemm_corrected" src="https://github.com/sarchlab/mgpusim/assets/2524066/e75aa2c4-a7ad-41b0-845e-4cf4bd8e8090">

@atyangshuang Would you be able to give us a pull request to get it addressed?

**Comment by yyssys** [comment]
Thank you for your answer. I am a beginner and I am still learning.
